cmd/gke-gcloud-auth-plugin/main.go

```diff
@@ -22,54 +26,81 @@ var (
 	//   email instead of numeric uniqueID.
 	defaultScopes = []string{
 		"https://www.googleapis.com/auth/cloud-platform",
-		"https://www.googleapis.com/auth/userinfo.email"}
+		"https://www.googleapis.com/auth/userinfo.email",
+	}
 )
 
-type credential struct {
-	AccessToken string                 `json:"access_token"`
-	TokenExpiry time.Time              `json:"token_expiry"`
-	X           map[string]interface{} `json:"-"` // Rest of the fields should go here.
-}
-
-// gcloudConfiguration is the struct unmarshalled
-// from gcloud config in json format
-type gcloudConfiguration struct {
-	Credential credential             `json:"credential"`
-	X          map[string]interface{} `json:"-"` // Rest of the fields should go here.
-}
+// types unmarshaled from gcloud config in json format
+type (
+	gcloudConfiguration struct {
```

Comment:
This looks pretty weird --- also do you need to preserve the "X" fields?  Or are unknown fields ignored.

Reply:
It's that or name a bunch of structs that we never refer to by name. What do you prefer?

There are no references to X. We don't rewrite the config json anywhere so there's not need to preserve.